operator/rust/crates/operator/src/start_operator.rs

```diff
@@ -140,118 +132,115 @@ async fn monitor_new_tasks() -> Result<()> {
-        .expect("wrong stake registry contract address");
-    let stake_registry_contract_address = Address::from_str(&STAKE_REGISTRY_CONTRACT_ADDRESS)
-        .expect("wrong stake registry contract address");
+async fn register_operator() -> Result<()> {
+    let signer = PrivateKeySigner::from_str(&KEY.clone())?;
+    let wallet = EthereumWallet::from(signer.clone());
+    let pr = ProviderBuilder::new()
+        .with_recommended_fillers()
+        .wallet(wallet)
+        .on_http(Url::from_str(&ANVIL_RPC_URL)?);
 
     let default_slasher = Address::ZERO; // We don't need slasher for our example.
     let default_strategy = Address::ZERO; // We don't need strategy for our example.
 
+    let data = std::fs::read_to_string("contracts/deployments/core/31337.json")?;
+    let el_parsed: EigenLayerData = serde_json::from_str(&data)?;
+    let delegation_manager_address: Address = el_parsed.addresses.delegation.parse()?;
+    let avs_directory_address: Address = el_parsed.addresses.avs_directory.parse()?;
+
     let elcontracts_reader_instance = ELChainReader::new(
-        logger.clone(),
+        get_logger().clone(),
         default_slasher,
-        delegation_manager_contract_address,
-        avs_directory_contract_address,
-        RPC_URL.clone(),
+        delegation_manager_address,
+        avs_directory_address,
+        ANVIL_RPC_URL.to_string(),
     );
     let elcontracts_writer_instance = ELChainWriter::new(
-        delegation_manager_contract_address,
+        delegation_manager_address,
         default_strategy,
+        Address::ZERO,
         elcontracts_reader_instance.clone(),
-        RPC_URL.clone(),
+        ANVIL_RPC_URL.to_string(),
         KEY.clone(),
     );
 
-    let operator = Operator::new(
-        wallet.address(),
-        wallet.address(),
-        Address::ZERO,
-        0u32,
-        None,
-    );
+    let operator = Operator {
+        address: signer.address(),
+        earnings_receiver_address: signer.address(),
+        delegation_approver_address: Address::ZERO,
+        staker_opt_out_window_blocks: 0u32,
+        metadata_url: None,
+    };
 
-    // let is_registered = elcontracts_reader_instance
-    //     .is_operator_registered(wallet.address())
-    //     .await
-    //     .unwrap();
-    // logger
-    //     .tracing_logger
-    //     .as_ref()
-    //     .unwrap()
-    //     .info(&format!("is registered {}", is_registered), &[""]);
-    #[allow(unused_doc_comments)]
-    ///In case you are running holesky. Comment the below register_as_operator call after the first
-    /// call . Since we can register only once per operator.
-    let _tx_hash = elcontracts_writer_instance
+    let is_registered = elcontracts_reader_instance
+        .is_operator_registered(signer.address())
+        .await
+        .unwrap();
+    get_logger().info(&format!("is registered {}", is_registered), &"");
+    #[allow(unused)]
+    let tx_hash = elcontracts_writer_instance
```

Comment:
It seems that we assume the function `.register_as_operator(operator)`  always succeeds. Should we integrate the return result for success?